### ISSUE TITLE
Fail to run an Kubernetes 1.22 with unexpected error while running the main control loop

### ISSUE THREAD

**Issue by petershaw** [nats-io/nats-operator#327]
Hi, 
I am updating to Kubernetes 1.22 and the nats operator fail to start: 
```
$  k logs nats-operator-5dcf9bd576-qg6wn
time="2021-08-19T09:56:49Z" level=info msg="nats-operator Version: 0.7.4"
time="2021-08-19T09:56:49Z" level=info msg="Git SHA: b96068c"
time="2021-08-19T09:56:49Z" level=info msg="Go Version: go1.15.1"
time="2021-08-19T09:56:49Z" level=info msg="Go OS/Arch: linux/amd64"
time="2021-08-19T09:56:49Z" level=warning msg="nats-operator is operating at the cluster scope (experimental)"
E0819 09:56:49.974721       1 event.go:296] Could not construct reference to: '&v1.Endpoints{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"nats-operator", GenerateName:"", Namespace:"nats-io", SelfLink:"", UID:"2178a147-e609-4484-9e7a-5d50354e25e9", ResourceVersion:"104236306", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63752633339, loc:(*time.Location)(0x1de7740)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string{"control-plane.alpha.kubernetes.io/leader":"{\"holderIdentity\":\"nats-operator-5dcf9bd576-qg6wn\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2021-08-19T09:51:08Z\",\"renewTime\":\"2021-08-19T09:56:49Z\",\"leaderTransitions\":2}"}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry{v1.ManagedFieldsEntry{Manager:"nats-operator", Operation:"Update", APIVersion:"v1", Time:(*v1.Time)(0xc0004e01c0), Fields:(*v1.Fields)(nil)}}}, Subsets:[]v1.EndpointSubset(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' 'LeaderElection' 'nats-operator-5dcf9bd576-qg6wn became leader'
time="2021-08-19T09:56:49Z" level=fatal msg="unexpected error while running the main control loop: the server could not find the requested resource"
```

**msg="unexpected error while running the main control loop: the server could not find the requested resource"**

Kubernetes: 
```
Client Version Major:"1", Minor:"22", GitVersion:"v1.22.0"
Server Version Major:"1", Minor:"22", GitVersion:"v1.22.0"
```

**Comment by invidian** [comment]
This is most likely caused by CRDs management using old API version, here: https://github.com/nats-io/nats-operator/blob/2b7a73f39af9a145a9b3801243298067a8e792d7/pkg/util/kubernetes/crd.go#L130

Returning unwrapper error here: https://github.com/nats-io/nats-operator/blob/eef5976979e500a8077724220abc5ea8c934d9db/pkg/controller/controller.go#L220

**Comment by variadico** [comment]
This PR updates the project to use the current CRD APIs: https://github.com/nats-io/nats-operator/pull/333

**Comment by variadico** [comment]
We now use Kubernetes 1.22 clients, as of https://github.com/nats-io/nats-operator/pull/336.
Closing for now, but feel free to reopen if you still have problems.

**Comment by petershaw** [comment]
is that fix incuded in the 0.8.0 docker? 
I ask, because i get the same error after updating to the latest docker image
```
time="2021-11-15T13:06:13Z" level=info msg="nats-operator Version: 0.8.0"
time="2021-11-15T13:06:13Z" level=info msg="Git SHA: b7b2800"
time="2021-11-15T13:06:13Z" level=info msg="Go Version: go1.15.1"
time="2021-11-15T13:06:13Z" level=info msg="Go OS/Arch: linux/amd64"
time="2021-11-15T13:06:13Z" level=warning msg="nats-operator is operating at the cluster scope (experimental)"
E1115 13:06:31.105421       1 event.go:296] Could not construct reference to: '&v1.Endpoints{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"nats-operator", GenerateName:"", Namespace:"nats-io", SelfLink:"", UID:"2178a147-e609-4484-9e7a-5d50354e25e9", ResourceVersion:"157687296", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63752633339, loc:(*time.Location)(0x1de9740)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string{"control-plane.alpha.kubernetes.io/leader":"{\"holderIdentity\":\"nats-operator-6c5875574c-wmlmw\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2021-11-15T13:06:31Z\",\"renewTime\":\"2021-11-15T13:06:31Z\",\"leaderTransitions\":6}"}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry{v1.ManagedFieldsEntry{Manager:"nats-operator", Operation:"Update", APIVersion:"v1", Time:(*v1.Time)(0xc0000b3960), Fields:(*v1.Fields)(nil)}}}, Subsets:[]v1.EndpointSubset(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' 'LeaderElection' 'nats-operator-6c5875574c-wmlmw became leader'
time="2021-11-15T13:06:31Z" level=fatal msg="unexpected error while running the main control loop: the server could not find the requested resource"
```

Thanks again.

**Comment by petershaw** [comment]
Works brilliant! Thanks!
